### PR TITLE
[py] Tidy up python exception handling

### PR DESCRIPTION
**Pull request by symonk**
Would like this for `4.3` not `4.2`.  A lot of the python code has some unnecessary exception handling catching some exception and instantly re-raising it; best to just let it bubble up here than catching it to begin with; I've removed usages of:

- catching an exception and instantly re-raising it with just not simply catching it
- `raise e` with just `raise` to avoid creating unnecessary convoluted stack traces

### CODE REVIEW

File: py/selenium/webdriver/remote/switch_to.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ def send_handle(h):
         try:
             # Try using it as a handle first.
             send_handle(window_name)
-        except NoSuchWindowException as e:
```
Comment (justnpT) [inline py/selenium/webdriver/remote/switch_to.py:144]: why are removing the "as e" ?

File: py/selenium/webdriver/support/color.py [hunk 2]
```diff
@@ -123,7 +123,7 @@ def hue_to_rgb(lum1: float, lum2: float, hue: float) -> float:
                     hue -= 1
 
                 if hue < 1.0 / 6.0:
-                    return (lum1 + (lum2 - lum1) * 6.0 * hue)
+                    return lum1 + (lum2 - lum1) * 6.0 * hue
```
Comment (justnpT) [inline py/selenium/webdriver/support/color.py:128]: nice

File: py/test/selenium/webdriver/common/webdriverwait_tests.py [hunk 3]
```diff
@@ -38,7 +38,7 @@ def test_should_fail_with_invalid_selector_exception(driver, pages):
         WebDriverWait(driver, 0.7).until(EC.presence_of_element_located((By.XPATH, "//*[contains(@id,'something'")))
 
 
-def test_should_explicitly_wait_for_asingle_element(driver, pages):
+def test_should_explicitly_wait_for_a_single_element(driver, pages):
```
Comment (justnpT) [inline py/test/selenium/webdriver/common/webdriverwait_tests.py:41]: nice !

File: py/selenium/webdriver/remote/switch_to.py (hunk 1 above)
Comment (symonk) [reply py/selenium/webdriver/remote/switch_to.py:144]: 0-arg raise restores the last handled exception (and traceback)

File: py/selenium/webdriver/remote/switch_to.py [hunk 4]
```diff
@@ -53,7 +53,7 @@ def alert(self) -> Alert:
                 alert = driver.switch_to.alert
         """
         alert = Alert(self._driver)
-        alert.text
+        _ = alert.text
```
Comment (AutomatedTester) [inline py/selenium/webdriver/remote/switch_to.py:56]: Why are we doing this?

**Comment by titusfortner** [comment]
fyi. I set this to draft since it says not to merge

File: py/selenium/webdriver/remote/switch_to.py (hunk 4 above)
Comment (symonk) [reply py/selenium/webdriver/remote/switch_to.py:56]: just to indicate the value is never used; will revert this one; edit: thoughts on keeping as-is? (most IDEs etc will complain about the statement with no effect)

**Comment by AutomatedTester** [comment]
@symonk why do you think we should wait until for 4.3? I can't see any breaking changes

**Comment by symonk** [comment]
> @symonk why do you think we should wait until for 4.3? I can't see any breaking changes

just wanted more time to test, but I think it's ok, was branched off trunk when flake8 was complaining; will update

**Review by AutomatedTester — APPROVED** [review]

**Pull request merged**